zk-token-sdk/src/instruction/fee_sigma.rs

```diff
@@ -0,0 +1,203 @@
+//! The fee sigma proof instruction.
+//!
+//! A fee sigma proof certifies that a Pedersen commitment to a transfer fee for SPL Token 2022 is
+//! well-formed.
+
+#[cfg(not(target_os = "solana"))]
+use {
+    crate::{
+        encryption::pedersen::{PedersenCommitment, PedersenOpening},
+        errors::ProofError,
+        sigma_proofs::fee_proof::FeeSigmaProof,
+        transcript::TranscriptProtocol,
+    },
+    merlin::Transcript,
+    std::convert::TryInto,
+};
+use {
+    crate::{
+        instruction::{ProofType, ZkProofData},
+        zk_token_elgamal::pod,
+    },
+    bytemuck::{Pod, Zeroable},
+};
+
+/// The instruction data that is needed for the `ProofInstruction::VerifyFeeSigma` instruction.
+///
+/// It includes the cryptographic proof as well as the context data information needed to verify
+/// the proof.
+#[derive(Clone, Copy, Pod, Zeroable)]
+#[repr(C)]
+pub struct FeeSigmaProofData {
+    pub context: FeeSigmaProofContext,
+
+    pub proof: pod::FeeSigmaProof,
+}
+
+/// The context data needed to verify a pubkey validity proof.
+#[derive(Clone, Copy, Pod, Zeroable)]
+#[repr(C)]
+pub struct FeeSigmaProofContext {
+    pub fee_commitment: pod::PedersenCommitment,
+
+    pub delta_commitment: pod::PedersenCommitment,
+
+    pub claimed_commitment: pod::PedersenCommitment,
```

Comment:
Some comments that drescribe what "fee", "delta", and "claimed" are would be useful here

Reply:
Agreed. Describing the delta and claimed fields are a bit involved, so I added a link to the ZK Token proof program docs for now, which will (soon) have a more formal description of these fields.